core/src/main/scala/kafka/log/Log.scala

```diff
@@ -1572,12 +1572,28 @@ class Log(@volatile var dir: File,
         val offsetIdxFile = offsetIndexFile(dir, newOffset)
         val timeIdxFile = timeIndexFile(dir, newOffset)
         val txnIdxFile = transactionIndexFile(dir, newOffset)
```

Comment:
Seems we only use these variables in the `else` case below. Perhaps we can move them there?